languages/python/atom/element.py

```
@@ -126,12 +135,33 @@ def __init__(self, name, host=ATOM_NUCLEUS_HOST, port=ATOM_NUCLEUS_PORT, metrics
         self._reference_get_metrics = None
 
         #
-        # Set up logging levels
+        # Set up metrics logging levels
         #
-        self._log_level = LogLevel[os.getenv("ATOM_LOG_LEVEL", "INFO")]
         self._metrics_level = MetricsLevel[os.getenv("ATOM_METRICS_LEVEL", "TIMING")]
         self._metrics_use_aggregation = os.getenv("ATOM_METRICS_USE_AGGREGATION", "FALSE") == "TRUE"
 
+        #
+        # Set up logger
+        # 
+        try: 
+            rfh = logging.handlers.RotatingFileHandler(f'{ATOM_LOG_DIR}{self.name}.log', maxBytes=ATOM_LOG_FILE_SIZE)
+            extra = {'element_name': self.name}
+            formatter = logging.Formatter("%(asctime)s element:%(element_name)s [%(levelname)s] %(message)s")
+            rfh.setFormatter(formatter)
+            logger.addHandler(rfh)
+            self.logger = logging.LoggerAdapter(logger, extra)
+        except Exception as e: 
```

Generic try/except is generally not a great strategy. We use it in certain places but I believe the better way to handle this is to catch the errors we know about/can reasonably happen explicitly and let it throw in the event we hit a new one.

One downside of this way of doing it is that you lose the full stack trace as well which is quite useful. 

Finally, putting this much code in a single `try/except` is also not ideal -- they should be small and specific to known errors in each case.